flow360/component/simulation/blueprint/codegen/generator.py

```diff
@@ -28,130 +31,246 @@ def _indent(code: str, level: int = 1) -> str:
     return "\n".join(spaces + line if line else line for line in code.split("\n"))
 
 
-def expr_to_code(expr: Any) -> str:
-    """Convert an expression model back to Python code."""
-    if expr is None:
+def check_syntax_type(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+        if result is None:
+            raise ValueError(f"Unsupported syntax type, available {[syntax.name for syntax in TargetSyntax]}")
+        return result
+    return wrapper
+
+
+@check_syntax_type
+def _empty(syntax):
+    if syntax == TargetSyntax.PYTHON:
         return "None"
```

Comment:
```suggestion
        return None
```
Seems check_syntax_type expects None instead.

Also why CPP and Python returns differently?

Reply:
Because the string representation of the null object is different. Now that I look I realize I made a mistake. It should either be `NULL` or `nullptr`

Reply:
`check_syntax_type` is just a decorator that will raise an error if the specified target syntax is invalid (its a bit hacky, as we always expect the functions to return a string representation of the expression so if we return `None` it probably means something has gone wrong.

Reply:
Changed to "nullptr" for CPP in [b7fbe78](https://github.com/flexcompute/Flow360/pull/1009/commits/b7fbe78d70b4085172cb2bb9cde85adf17cd1b68)